### ISSUE TITLE
Show when an image is being uploaded

### ISSUE THREAD

**Issue by MehediH** [thepracticaldev/dev.to#7907]
**Is your feature request related to a problem? Please describe.**

Currently, the image uploader does not display when an image is being uploaded. This can cause confusion on a slow network or when uploading large files, as it is not clear when the image is being uploaded. The app currently shows when an error occurs, but there is no UI element indicating when the image is being uploaded. I have personally experienced this issue, where I was left wondering if my images are being uploaded or not when creating a new post on the site. 

**Describe the solution you'd like**

The ideal interface I have considered would be to combine the error message element and the update status into a single element above the "Done" button. In this implementation, the app would first show "Uploading images..." when the user uploads an image and it passes the validations, and if it succeeds, they will be shown an "Upload successful" message. When there is an error, it would simply show the error message instead of the upload status:

![image](https://user-images.githubusercontent.com/10274434/82163336-6d0e5080-98a2-11ea-9df3-22ce178bc67e.png)

**Describe alternatives you've considered**

The alternative solution would be to introduce a simple text element above the "Done" button that says something like "Uploading images..." while the image is being uploaded. This would still mean that there will be a separate element for the upload errors (as it is now) on the top, which I feel like is not the best UI. 

**Additional context**

N/A

**Solution**

I have implemented the solution locally and would be happy to create a pull request if this is considered a real problem.

**Comment by rhymes** [comment]
Thanks for the issue! We'll take your request into consideration and follow up if we decide to tackle this issue.

To our amazing contributors: issues labeled `type: bug` are always up for grabs, but for feature requests, please wait until we add a `ready for dev` before starting to work on it.

To claim an issue to work on, please leave a comment. If you've claimed the issue and need help, please ping @thepracticaldev/oss and we will follow up within 3 business days.

For full info on how to contribute, please check out our [contributors guide](https://github.com/thepracticaldev/dev.to/blob/master/CONTRIBUTING.md).

**Comment by ankiiitraj** [comment]
Hi, I am new here and I would love to take on the issue

**Comment by MehediH** [comment]
> Hi, I am new here and I would love to take on the issue

@ankiiitraj Hi Ankit! I already have a fix for the issue and just waiting for the team to give me a go ahead on this before sending a PR.

**Comment by ankiiitraj** [comment]
I am sorry, I didn't read the Solution part of you description.

**Comment by rhymes** [comment]
@MehediH what feedback do you need?

**Comment by rhymes** [comment]
Sorry I didn't notice the last bit of the issue description :D

Anyway, all bugs don't need approval for a fix, in this case though we might want to hold off as we're completely redoing the editor: https://github.com/thepracticaldev/dev.to/pull/7524#issuecomment-634719256

Sorry for that @MehediH :-(

**Comment by MehediH** [comment]
@rhymes I thought I had to wait for approval from the team before I went ahead and submitted a PR. Regarding the editor refactor, the v2 editor does not have a fix for this issue, and my fix actually works for both v2 and the current version of the editor.

Let me know if I should still hold off.

**Comment by rhymes** [comment]
@MehediH if you look at https://github.com/thepracticaldev/dev.to/pull/7524 the upload interface is quite different and yes, it's still not going to have this but at least we won't have to re-do this feature twice. so maybe hold off for that PR?

Thank you a lot BTW! :-)

**Comment by MehediH** [comment]
Okay cool, I will wait! Thank you :)

**Comment by atsmith813** [comment]
There is a size limit of 25mb on images and 50mb on videos:

https://github.com/thepracticaldev/dev.to/blob/71fba61ae26acdecc6f9026389403bf93407aa44/app/javascript/packs/validateFileInputs.js#L16-L19

**Comment by Kvaibhav01** [comment]
@atsmith813 okay I get it. What about GIF? Mine is of 7.47 MB and I get that error while uploading this GIF.

**Comment by atsmith813** [comment]
GIF would be considered under images here. However, we have a few existing issues around GIFs 🙈 . Here's the most recent one for example:

https://github.com/thepracticaldev/dev.to/issues/7889

I would try reducing your GIF size below 5 MB and/or checking the frame rate. That has worked for me in the past. If that still doesn't work, then it's a symptom of the outstanding issue.

> Inline Images
When adding GIFs to posts and comments, please note that there is a limit of 200 megapixels per frame/page.

**Comment by Kvaibhav01** [comment]
Okay, thanks! I'll try to reduce the size and will get back how it works. :)

**Comment by Kvaibhav01** [comment]
I stripped down the size from 7.47 to 3.16 MB. Still, I get the error:

> A server error has occurred!

In the console I get this:

![Error from comsole](https://user-images.githubusercontent.com/11731837/84425345-067d1800-ac3f-11ea-9edf-2855aaf60892.png)

**Comment by atsmith813** [comment]
That looks like it's related to our existing GIF issues.

**Comment by nickytonline** [comment]
#8986 implements this for the body content images, but not cover image. I will have another PR coming up for that today or tomorrow.
